maven/tests/com.liferay.ide.maven.core.tests/src/com/liferay/ide/maven/core/tests/NewLiferayJSFModuleProjectOpTests.java

```diff
@@ -0,0 +1,228 @@
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.maven.core.tests;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.liferay.ide.core.ILiferayProjectImporter;
+import com.liferay.ide.core.LiferayCore;
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.core.util.ZipUtil;
+import com.liferay.ide.project.core.jsf.NewLiferayJSFModuleProjectOp;
+import com.liferay.ide.project.core.jsf.NewLiferayJSFModuleProjectOpMethods;
+import com.liferay.ide.project.core.tests.ProjectCoreBase;
+
+import java.io.File;
+import java.net.URL;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.IncrementalProjectBuilder;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.sapphire.DefaultValueService;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.platform.ProgressMonitorBridge;
+import org.eclipse.wst.validation.internal.operations.ValidatorManager;
+import org.junit.Test;
+
+/**
+ * @author Simon Jiang
+ */
+@SuppressWarnings( "restriction" )
+public class NewLiferayJSFModuleProjectOpTests extends ProjectCoreBase
+{
+
+    @Test
+    public void testNewLiferayJSFModuleProjectOpProjectName() throws Exception
+    {
+        NewLiferayJSFModuleProjectOp op = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        op.setJsfProjectProvider( "maven-jsf" );
+        op.setProjectName( "Test1" );
+        Status projectNameOkValidationStatus1 = op.getProjectName().validation();
+        assertEquals( projectNameOkValidationStatus1.message(), "ok" );
+        op.setJsfProjectProvider( "maven-jsf" );
+        op.setProjectName( "#Test1" );
+        Status projectNameErrorValidationStatus = op.getProjectName().validation();
+        assertEquals( projectNameErrorValidationStatus.message(), "The project name is invalid." );
+        op.setJsfProjectProvider( "maven-jsf" );
+        op.setProjectName( "Test1_Abc" );
+        Status projectNameOkValidationStatus2 = op.getProjectName().validation();
+        assertEquals( projectNameOkValidationStatus2.message(), "ok" );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleProjectOpProjectExisted() throws Exception
+    {
+        NewLiferayJSFModuleProjectOp existedProjectop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        existedProjectop.setJsfProjectProvider( "gradle-jsf" );
```

Comment:
We should test more maven provider types than gradle since maven will be more widely used by JSF users.